source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -364,6 +374,54 @@ The annotations in the example POJO specify the following behavior:
 - Omit the ``relatedItems`` field and value when converting data
 - Use the ``Product(String name)`` constructor when instantiating the POJO
 
+
+.. _bsonextraelements-annotation-code-example:
+
+BsonExtraElements Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``@BsonExtraElements`` annotation allows you to specify a field to
+deserialize data from a MongoDB document that lacks a corresponding POJO
+field mapping. Consider a situation in which your application needs to handle
+data that conforms to different data schema versions. When your POJO fields
+change to accommodate the schema, you can continue to read, but not
+update, the remainder of the MongoDB document data from the field that uses
+this annotation.
+
+Consider a situation in which you need to update the ``Product`` POJO from the
```

Comment:
This is a great example, and actually one I hadn't thought of.  I think the more common case though is where you have a semi-structured schema where you want to model the structured part as POJO fields while still being able to read/update the unstructured part as a Document.  Perhaps we could add an example for that as well, or at least mention the use case?

Reply:
Based on our conversation, I'm planning to change the situation to talk about a situation in which you have a partially-structured schema.

I'd like any input you have on which example you think would be most suitable:

1) Once the BsonExtraElements field is established in the POJO, update the document and application's usage of that data without having to change the serial version id. E.g. Add a "hidden: true" field/value to the document to instruct the application from surfacing that POJO from an API call.

2) Lazy data migration: the goal is to get all the data to resemble the new schema reflected in the POJO, but it's updated only when you access the document rather than in bulk. Not common, but have encountered this situation before.

3) Overload application models to contain related metadata that other processes generate (such as search weights/relevance and ML data). Doesn't provide a very strong case for using the field unless the application needed to surface that data.